### PR TITLE
Wrong tensor size in quadrature

### PR DESCRIPTION
**Pull request by mscroggs**
In `quadrature::compute_gauss_jacobi_points`, the number of dimensions of a tensor was incorrect. A few other changes:

- ~Moved the -O2 to all builds, so that Release is sped up.~
- Added `-DCMAKE_BUILD_TYPE=Release` to the install instructions

### CODE REVIEW

File: .github/workflows/pythonapp.yml [hunk 1]
```diff
@@ -72,3 +72,34 @@ jobs:
           git add --all
           git commit --allow-empty -m "C++ FEniCS/basix@${{ github.sha }}"
           git push
+
+  build-not-dev:
```
Comment (garth-wells) [inline .github/workflows/pythonapp.yml:76]: What flags are different for this build case? Ideally we'd pick up these issues on a single build and run.

Comment (mscroggs) [reply .github/workflows/pythonapp.yml:76]: I doesn't have `-DCMAKE_BUILD_TYPE=Developer`. For reasons beyond me some errors were only happening when building without developer mode (see eg https://github.com/bempp/bempp-cl/runs/2110771847)

Comment (garth-wells) [reply .github/workflows/pythonapp.yml:76]: The CMake build type sets the flags - need to understand which compiler flags need to be set.

**Comment by chrisrichardson** [comment]
Can we merge this, to fix the stack smashing?

**Comment by mscroggs** [comment]
Seems sensible. I don't think it will cause any conflicts with the xtensor branch

**Comment by garth-wells** [comment]
The changes to the CMake file are confusing to me - the logic is hard to follow. Seems wrong to force the optimisation level in debug mode.

**Comment by mscroggs** [comment]
Agreed. I think it makes sense to have `-O2` on Release and Developer but not Debug. Just pushed a change to do this
